### PR TITLE
Update default for span limits to 128, 1000 seems off

### PR DESCRIPTION
**Pull request by bogdandrutu**
128 is based on the value used in OpenCensus and it was not a problem there. See https://github.com/census-instrumentation/opencensus-java/blob/3d7f0b511dea392bbf68ee5268dd95857552dd87/api/src/main/java/io/opencensus/trace/config/TraceParams.java#L36

The limits for attributes and links were smaller, but I applied a max(of_all) function.

### CODE REVIEW

**Review by yurishkuro — APPROVED** [review]

File: specification/sdk-environment-variables.md [hunk 1]
```diff
@@ -77,9 +77,9 @@ Depending on the value of `OTEL_TRACES_SAMPLER`, `OTEL_TRACES_SAMPLER_ARG` may b
 
 | Name                            | Description                          | Default | Notes |
 | ------------------------------- | ------------------------------------ | ------- | ----- |
-| OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT | Maximum allowed span attribute count | 1000    |       |
-| OTEL_SPAN_EVENT_COUNT_LIMIT     | Maximum allowed span event count     | 1000    |       |
-| OTEL_SPAN_LINK_COUNT_LIMIT      | Maximum allowed span link count      | 1000    |       |
+| OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT | Maximum allowed span attribute count | 128     |       |
```
Comment (anuraaga) [inline specification/sdk-environment-variables.md:80]: Do we need to reset the compliance matrix for these variables?

Comment (bogdandrutu) [reply specification/sdk-environment-variables.md:80]: Is it an entry there?

Comment (bogdandrutu) [reply specification/sdk-environment-variables.md:80]: Oh, reset the `x`es

Comment (bogdandrutu) [reply specification/sdk-environment-variables.md:80]: I will do it in a followup PR, or after I merge #1420

**Review by anuraaga — APPROVED** [review]

**Review by arminru — APPROVED** [review]

**Comment by Oberon00** [comment]
I can tell you where the 1000 comes from: https://github.com/open-telemetry/opentelemetry-specification/issues/182#issuecomment-690883490

It was chosen because

> And to @arminru's point,we just choose a number that's well beyond 99.99% of use cases. Don't want to throw numbers out without data, but 1000 seems well above the needed amount.
-- @toumorokoshi


128 was also discussed as limit but thrown out because it might prevent valid use cases, see https://github.com/open-telemetry/opentelemetry-specification/issues/182#issuecomment-690285768

As this is only an experimental part of the spec, I think it's OK to rush a change here, but if combined with #1416, I would oppose it.

**Comment by arminru** [comment]
I still think 1000 is a good default to allow for 99.9% of all use cases _if this is not configurable_. With #1416 making configurability a MUST, however, going for a default of 128 should be fine since users have the option to adapt it if they're doing something special that would require them to go beyond that.

**Review by Oberon00 — APPROVED** [review]
I see. I am a bit confused then too why this became 1000, given that the 1000 was based on the assumption that it is *not* configurable while the 1000 appears exactly as default value for a *configuration* option.

**Comment by toumorokoshi** [comment]
> I see. I am a bit confused then too why this became 1000, given that the 1000 was based on the assumption that it is not configurable while the 1000 appears exactly as default value for a configuration option.

IIRC the original PR made span limits optionally configurable, and stated that the default or unconfigurable value is 1000.

FWIW 1000 was chosen over 128 because 128 could be ambiguous as some code-level limitation (e.g. right above an 8-bit signed integer), and 1000 is clearly a human-defined limit.

I'm not really sure why 128 vs 1000 (or visa versa) necessitates a change. but not here to debate the change, just add clarity.

**Pull request merged**